compiler/noirc_frontend/src/elaborator/unquote.rs

```diff
@@ -0,0 +1,306 @@
+        self.find_unquoted_exprs_in_expr(&mut index.index, unquoted_exprs);
+    }
+
+    fn find_unquoted_exprs_in_call(
+        &mut self,
+        call: &mut CallExpression,
+        unquoted_exprs: &mut Vec<ExprId>,
+    ) {
+        self.find_unquoted_exprs_in_expr(&mut call.func, unquoted_exprs);
+
+        for arg in &mut call.arguments {
+            self.find_unquoted_exprs_in_expr(arg, unquoted_exprs);
+        }
+    }
+
+    fn find_unquoted_exprs_in_method_call(
+        &mut self,
+        call: &mut MethodCallExpression,
+        unquoted_exprs: &mut Vec<ExprId>,
+    ) {
+        self.find_unquoted_exprs_in_expr(&mut call.object, unquoted_exprs);
+
+        for arg in &mut call.arguments {
+            self.find_unquoted_exprs_in_expr(arg, unquoted_exprs);
+        }
+    }
+
+    fn find_unquoted_exprs_in_constructor(
+        &mut self,
+        constructor: &mut ConstructorExpression,
+        unquoted_exprs: &mut Vec<ExprId>,
+    ) {
+        for (_, field) in &mut constructor.fields {
+            self.find_unquoted_exprs_in_expr(field, unquoted_exprs);
+        }
+    }
+
+    fn find_unquoted_exprs_in_access(
+        &mut self,
+        member_access: &mut MemberAccessExpression,
+        unquoted_exprs: &mut Vec<ExprId>,
+    ) {
+        self.find_unquoted_exprs_in_expr(&mut member_access.lhs, unquoted_exprs);
+    }
+
+    fn find_unquoted_exprs_in_cast(
+        &mut self,
+        cast: &mut CastExpression,
+        unquoted_exprs: &mut Vec<ExprId>,
+    ) {
+        self.find_unquoted_exprs_in_expr(&mut cast.lhs, unquoted_exprs);
+    }
+
+    fn find_unquoted_exprs_in_infix(
+        &mut self,
+        infix: &mut InfixExpression,
+        unquoted_exprs: &mut Vec<ExprId>,
+    ) {
+        self.find_unquoted_exprs_in_expr(&mut infix.lhs, unquoted_exprs);
+        self.find_unquoted_exprs_in_expr(&mut infix.rhs, unquoted_exprs);
+    }
+
+    fn find_unquoted_exprs_in_if(
+        &mut self,
+        if_: &mut IfExpression,
+        unquoted_exprs: &mut Vec<ExprId>,
+    ) {
+        self.find_unquoted_exprs_in_expr(&mut if_.condition, unquoted_exprs);
+        self.find_unquoted_exprs_in_expr(&mut if_.consequence, unquoted_exprs);
+
+        if let Some(alternate) = if_.alternative.as_mut() {
+            self.find_unquoted_exprs_in_expr(alternate, unquoted_exprs);
+        }
+    }
+
+    fn find_unquoted_exprs_in_tuple(
+        &mut self,
+        tuple: &mut [Expression],
+        unquoted_exprs: &mut Vec<ExprId>,
+    ) {
+        for field in tuple {
+            self.find_unquoted_exprs_in_expr(field, unquoted_exprs);
+        }
+    }
+
+    fn find_unquoted_exprs_in_lambda(
+        &mut self,
+        lambda: &mut Lambda,
+        unquoted_exprs: &mut Vec<ExprId>,
+    ) {
+        self.find_unquoted_exprs_in_expr(&mut lambda.body, unquoted_exprs);
+    }
+
+    /// Elaborate & store the unquoted expression in the given vector, then
```

Comment:
```suggestion
    /// Elaborate and store the unquoted expression in the given vector, then
```
nitpick, I prefer we write out `and` here